src/main/java/ch/njol/skript/expressions/ExprBossBarFlags.java

```diff
@@ -0,0 +1,115 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.expressions.base.PropertyExpression;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser;
+import ch.njol.util.Kleenean;
+import org.bukkit.boss.BarFlag;
+import org.bukkit.boss.BossBar;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+import java.util.ArrayList;
+import java.util.List;
+
+
+@Name("Boss Bar Flags")
+@Description("The flags of a bossbar. These flags control the behavior of the bossbar.")
+@Examples({"add darken sky to flags of player's bossbar"})
+@Since("INSERT VERSION")
+public class ExprBossBarFlags extends PropertyExpression<BossBar, BarFlag> {
+
+	static {
+		register(ExprBossBarFlags.class, BarFlag.class, "flags", "bossbars");
+	}
+
+	@SuppressWarnings("unchecked")
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, SkriptParser.ParseResult parseResult) {
+		setExpr((Expression<? extends BossBar>) exprs[0]);
+		return true;
+	}
+
+	@Override
+	@Nullable
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		switch (mode) {
+			case SET:
+			case ADD:
+			case REMOVE:
+			case DELETE:
+				return new Class[] {BarFlag.class};
+			default:
+				return null;
+		}
+	}
+
+	@Override
+	public void change(Event event, @Nullable Object[] delta, ChangeMode mode) {
+		for (BossBar bossBar : getExpr().getArray(event))
+			switch (mode) {
+				case SET:
+					clearFlags(bossBar);
+				case ADD:
+					for (Object flag : delta)
+						bossBar.addFlag((BarFlag) flag);
+					break;
+				case REMOVE:
+					for (Object flag : delta)
+						bossBar.removeFlag((BarFlag) flag);
+					break;
+				case DELETE:
+					clearFlags(bossBar);
+			}
+	}
+
+	@Override
+	protected BarFlag[] get(Event event, BossBar[] source) {
```

Comment:
get method comes before change and acceptChange in an expression

`SimpleExpression: init -> get (getAll) -> (acceptChange) -> (change) -> isSingle -> getReturnType -> toString`

https://github.com/SkriptLang/Skript/blob/master/code-conventions.md